x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/session/EqlSession.java

```diff
@@ -121,7 +120,6 @@ public void analyzedPlan(LogicalPlan parsed, ActionListener<LogicalPlan> listene
             indexWildcard,
             configuration.indicesOptions(),
             configuration.runtimeMappings(),
-            Version.CURRENT,
```

Comment:
Instead of doing the filter manually, filter the IndexResult directly:
```
map(listener, r -> preAnalyzer.preAnalyze(parsed, IndexCompatibility.compatible(indexResolution.get(), Version.fromId(configuration.version().id));
```

This way the rules that call IndexResolution.get() will get the index already filtered - since I cannot think of an instance where that is needed, I think that's fine.

Reply:
Maxes more sense indeed, fixed.